### ISSUE TITLE
modeord flag

### ISSUE THREAD

**Issue by unalmis** [flatironinstitute/jax-finufft#155]
Can we add the `modeord` flag [here](https://github.com/flatironinstitute/jax-finufft/blob/fecb33e594c18df71647b135e3d419a0fb47d793/src/jax_finufft/ops.py#L44)?

Ideally, this would pass the flag to finnufft, but a basic implementation is just

```python
if modeord == 1:
    source = jnp.fft.fftshift(source, axes=-1)
```

**Comment by unalmis** [comment]
Do you mean leading dimensions? To be consistent with jnp.vectorize all dimensions but the last one must broadcast. So I see how I could add an additional leading dimension to the source, but an API like FINUFFT where I can call nufft multiple times with the same plan would be convenient because to avoid materializing a tensor by concatenations source arrays.

**Comment by unalmis** [comment]
> > Is native numpy style vectoriziation supported

> Yes. Any number of leading batch dimensions are supported, and will be appropriately broadcasted. Give it a try!

@dfm `jax-finufft` does NOT support numpy-style vectorization. 

You can see here that jnp.vectorize(nufft2) works correctly, but native numpy vectorization is not supported.

https://github.com/PlasmaControl/DESC/blob/b52ad96cb8aa91bd210568c5f6a41a2efcb05ef4/tests/test_interp_utils.py#L258

**Comment by unalmis** [comment]
The issue is that the batch axis should be added at the front [here](https://github.com/flatironinstitute/jax-finufft/blob/fd9778497db0d410416b250ec6d6b43cbe611230/src/jax_finufft/shapes.py#L42) if `jax-finufft` is to support numpy style vectorization. It would be helpful to the users if the style of native vectorization that is supported is stated in the documentation.
